### DISCUSSION TITLE
Unable to install on Laravel:10.10 & Livewire:3

### DISCUSSION THREAD

**Discussion by ahtesham-clcbws** [rappasoft/laravel-livewire-tables#1332] [General]
### What happened?

unable to install, composer through errors about livewire version, I already install and build my half project onto livewire, but when i downgraded, livewire throught errors.

### PHP Version

8.1.x

### Laravel Version

10.10

### Alpine Version

3.13

### Theme

Tailwind 3.x

### Notes

MY COMPOSER PACKAGES BELOW:-

    "require": {
        "php": "^8.1",
        "guzzlehttp/guzzle": "^7.2",
        "laravel/framework": "^10.10",
        "laravel/jetstream": "^4.0",
        "laravel/sanctum": "^3.2",
        "laravel/tinker": "^2.8",
        "livewire/livewire": "*",
        "spatie/laravel-activitylog": "^4.7",
        "spatie/laravel-backup": "^8.3",
        "spatie/laravel-health": "^1.23",
        "spatie/laravel-medialibrary": "*",
        "spatie/laravel-model-states": "^2.4",
        "spatie/laravel-notification-log": "^1.1",
        "spatie/laravel-permission": "^5.10",
        "spatie/laravel-tags": "^4.5"
    },
    "require-dev": {
        "fakerphp/faker": "^1.9.1",
        "laravel/pint": "^1.0",
        "laravel/sail": "^1.18",
        "laravel/telescope": "^4.16",
        "mockery/mockery": "^1.4.4",
        "nunomaduro/collision": "^7.0",
        "phpunit/phpunit": "^10.1",
        "spatie/laravel-ignition": "^2.0"
    },

### Error Message

 Problem 1
    - rappasoft/laravel-livewire-tables[v0.1.0, ..., v0.2.1] require php ^7.2.5 -> your php version (8.1.10) does not satisfy that requirement.
    - rappasoft/laravel-livewire-tables[v0.3.0, ..., v0.3.2] require php ^7.3 -> your php version (8.1.10) does not satisfy that requirement.
    - rappasoft/laravel-livewire-tables[v0.3.3, ..., v0.4.0] require livewire/livewire ^1.3|^2.0 -> found livewire/livewire[v1.3.0, ..., v1.3.9, v2.0.0, ..., v2.12.6] but the package is fixed to v3.0.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - rappasoft/laravel-livewire-tables[v1.11.0, ..., v1.14.0] require livewire/livewire ^2.5.3 -> found livewire/livewire[v2.5.3, ..., v2.12.6] but the package is fixed to v3.0.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - rappasoft/laravel-livewire-tables v1.15.0 requires livewire/livewire ^2.6.2 -> found livewire/livewire[v2.6.2, ..., v2.12.6] but the package is fixed to v3.0.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - rappasoft/laravel-livewire-tables v1.16.0 requires livewire/livewire ^2.6.5 -> found livewire/livewire[v2.6.5, ..., v2.12.6] but the package is fixed to v3.0.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - rappasoft/laravel-livewire-tables[v1.0.0, ..., v1.19.1] require livewire/livewire ^2.0 -> found livewire/livewire[v2.0.0, ..., v2.12.6] but the package is fixed to v3.0.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - rappasoft/laravel-livewire-tables[v1.19.2, ..., v1.25.1, v2.0.0, ..., v2.15.0] require livewire/livewire ^2.6 -> found livewire/livewire[v2.6.0, ..., v2.12.6] but the package is fixed to v3.0.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires rappasoft/laravel-livewire-tables * -> satisfiable by rappasoft/laravel-livewire-tables[v0.1.0, ..., v0.4.0, v1.0.0, ..., v1.25.1, v2.0.0, ..., v2.15.0].

**Comment by quentindecosne** [comment]
You need to install v3

`composer require "rappasoft/laravel-livewire-tables":"dev-v3-master"`

**Comment by lrljoe** [comment]
Keep in mind that v3 is in beta, so recommendation is not to use in production.

Once any major bugs are ironed out then it'll be released.

**Comment by AleAtSoftplace** [comment]
The install is getting stuck because you can't upgrade to Livewire 3 without upgrading the Datatables plugin.
The full command would be:
`
composer require livewire/livewire "rappasoft/laravel-livewire-tables":"dev-v3-master" -W`

**Comment by lrljoe** [comment]
@ahtesham-clcbws - how are you doing with this now, if you're still stuck, then jump on the Discord and I'll try to lend a hand!

**Comment by ahtesham-clcbws** [comment]
Hi,

I am able to install the package from your comments, but for stuck on some errors, like when I change the columns from show/hide in UI, it will crash 50/50 chances, maybe there is issues like I don't understand the code or something, but try to figure out.
BTW, i would really like to join you on discord.
Let me know your handle

**Comment by lrljoe** [comment]
[https://discord.gg/mkmEfq7hUC](https://discord.gg/mkmEfq7hUC) is the official Discord for the package. I'm there fairly consistently as is Anthony from Rappasoft, and questions get answered fairly swiftly by us or other knowledgeable users :)

Does your Column Select issue look the same as:
[https://github.com/rappasoft/laravel-livewire-tables/issues/1318
](https://github.com/rappasoft/laravel-livewire-tables/issues/1318
)

**Comment by ahtesham-clcbws** [comment]
Yeas something like that

**Comment by lrljoe** [comment]
I'm struggling to replicate it on Tailwind, but have been able to on Bootstrap (which I'm working on a fix for)

Feel free to join the Discord to share further info btw, I do pay more attention to Discord than Github at the moment

**Comment by lrljoe** [comment]
May be worth trying the develop branch in the interim too:

```
composer require "rappasoft/laravel-livewire-tables":"dev-v3-develop"
```

This has the latest fixes etc
